### DISCUSSION TITLE
Updating current/live NixOS WSL Instance

### DISCUSSION THREAD

**Discussion by moekhalil** [nix-community/NixOS-WSL#407] [Q&A]
Previously (a few months back) I installed NixOS on my Windows 11 WSL using this repo's included instructions. I have a working running NixOS distribution running on WSL. I have updated nixos itself to the latest 23.11, but I noticed that `nixos-wsl` folder inside `/etc/nixos/` is outdated, including some of the files in `modules` that `nixos-rebuild <switch|build>` was complaining about because it references a `types.string` type instead of `str`.

I went ahead and updated the file (`/etc/nixos/nixos-wsl/modules/wsl-distro.nix`), and the error went away, but I noticed that that file is outdated compared to the latest version in this repo, as that has been fixed a while back. How can I update the whole `nixos-wsl` folder and nested folders like `modules` to better line-up with the current main branch in this repo? 

**TL-DR:** How can I update the "**NixOS-WSL**" part of my NixOS distro?

**Comment by nzbr** [comment]
Please refer to the of the "Migrating from 22.05 releases" section of the [23.5.5.0 Release Notes](https://github.com/nix-community/NixOS-WSL/releases/tag/23.5.5.0). You don't need to downgrade your `nixos` channel, obviously. Just keep it at 23.11

**Comment by moekhalil** [comment]
@nzbr I appreciate the follow up and apologize for not following up here, as I did see that after posting this discussion, and the instructions worked perfectly!

BTW, thank you for all your very helpful and awesome contributions - both the code and documentation!

Closing discussion.
